yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,184 @@
+            'description': str,
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'alt-title': None,
+            'uploader': str,
+            'channel': str,
+            'uploader_id': str,
+            'channel_id': str,
+            'uploader_url': r're:^https?://.*',
+            'channel_url': r're:^https?://.*',
+            'timestamp': int,
+            'like_count': int,
+            'repost_count': int,
+            'comment_count': int,
+            'webpage_url': r're:^https?://.*',
+            'tags': list,
+            'subtitles': dict,
+            'comments': list,
+        },
+    }]
+
+    def traverse_replies(self, thread_node, root_uri):
+        parent_uri = traverse_obj(thread_node, ('post', 'record', 'reply', 'parent', 'uri'))
+        parent_id = 'root' if parent_uri == root_uri else parent_uri
+        author_handle = traverse_obj(thread_node, ('post', 'author', 'handle'))
+        author_did = traverse_obj(thread_node, ('post', 'author', 'did'))
+
+        formatted_comments = [{
+            'id': traverse_obj(thread_node, ('post', 'uri')),
+            'text': traverse_obj(thread_node, ('post', 'record', 'text')),
+            'timestamp': parse_iso8601(traverse_obj(thread_node, ('post', 'record', 'createdAt'))),
+            'parent': parent_id,
+            'like_count': traverse_obj(thread_node, ('post', 'likeCount')),
+            'author': traverse_obj(thread_node, ('post', 'author', 'displayName')),
+            'author_id': author_did,
+            'author_thumbnail': traverse_obj(thread_node, ('post', 'author', 'avatar'), expected_type=url_or_none),
+            'author_url': f'https://bsky.app/profile/{author_handle}',
+            'author_is_uploader': 'Yes' if author_did in root_uri else 'No',
+        }]
+
+        if replies := thread_node.get('replies'):
+            for reply in replies:
+                formatted_comments.extend(self.traverse_replies(reply, root_uri))
+        return formatted_comments
+
+    def _real_extract(self, url):
+        handle, video_id = self._match_valid_url(url).groups()
+        did = self._download_json(
+            'https://bsky.social/xrpc/com.atproto.identity.resolveHandle',
+            video_id, query={'handle': handle}, expected_status=200).get('did')
+
+        meta = self._download_json(
+            'https://public.api.bsky.app/xrpc/app.bsky.feed.getPostThread',
+            video_id, headers={'Content-Type': 'application/json'},
+            query={'uri': f'at://{did}/app.bsky.feed.post/{video_id}',
+                   'depth': 6 if self.get_param('write_comments') else 0, 'parentHeight': 0},
+            expected_status=200)
+
+        formats, subs = self._extract_m3u8_formats_and_subtitles(
+            traverse_obj(meta, ('thread', 'post', 'embed', 'playlist')),
+            video_id, 'mp4', 'm3u8_native', m3u8_id='hls', fatal=False,
+            note='Downloading HD m3u8 information', errnote='Unable to download HD m3u8 information')
```

Comment:
This does not extract the best available format.

For example for https://bsky.app/profile/blu3blue.bsky.social/post/3l4omssdl632g, the best format at 634x1080 resolution can be downloaded via
```
https://bsky.social/xrpc/com.atproto.sync.getBlob?did=did:plc:pzdr5ylumf7vmvwasrpr5bf2&cid=bafkreiamqpagnuwf72e5zj2xypoew3t2arxho4nse3le5l4zwou4hcrfne
```

See https://github.com/mikf/gallery-dl/commit/7d6520e15dba1a50e7c3609293efe3ddd521aa75 or https://github.com/danbooru/danbooru/pull/5816/commits/dc37dd19ee23bddab34f3d4b9c707b8dd898871b

Reply:
Thanks! Implemented in 8c1dd6c4dabdabd2e8418d3dfcfc880e37a7a49c. :)

Reply:
I think you are supposed to extract *all* available formats for a yt-dlp extractor and not just the one with the best quality, so `_extract_m3u8_formats_and_subtitles` results plus blob.

Reply:
Ah right, of course. Thank you! c17c3e5d52d6de14bb2d0960654a89449bb101ea